src/combinatorics/simple_graph/basic.lean

```diff
@@ -281,9 +281,51 @@ def edge_set : set (sym2 V) := sym2.from_rel G.symm
 
 @[simp] lemma mem_edge_set : ⟦(v, w)⟧ ∈ G.edge_set ↔ G.adj v w := iff.rfl
 
+lemma not_is_diag_of_mem_edge_set {e : sym2 V} (h : e ∈ G.edge_set) : ¬ e.is_diag :=
+sym2.ind (λ v w, adj.ne) e h
+
+@[simp] lemma edge_set_sdiff (G H : simple_graph V) :
+  (G \ H).edge_set = G.edge_set \ H.edge_set :=
+by { ext ⟨x,y⟩, exact ⟨id, id⟩ }
+
+@[simp] lemma edge_set_inf (G H : simple_graph V) :
+  (G ⊓ H).edge_set = G.edge_set ∩ H.edge_set :=
+by { ext ⟨x,y⟩, exact ⟨id, id⟩ }
+
+@[simp] lemma edge_set_sup (G H : simple_graph V) :
+  (G ⊔ H).edge_set = G.edge_set ∪ H.edge_set :=
+by { ext ⟨x,y⟩, exact ⟨id, id⟩ }
+
+@[simp] lemma edge_set_sdiff_sdiff_is_diag (G : simple_graph V)  (s : set (sym2 V)) :
```

Comment:
This lemma might seem weird, so it might be worth mentioning in the documentation that it's a `simp` lemma so that `(G \ from_edge_set s).edge_set = G.edge_set \ s` can be proved by `simp`.

Reply:
"Mentioning in the doc" simply means commenting the lemma à la `/- blah -/` or is it `/-- blah -/` ?

Reply:
Good question, it could go either way. Doesn't hurt to have it be `/-- ... -/`. (By the way, I just noticed you have two spaces between the arguments rather than one.)